Modules/Common/src/TH2XlineReductor.cxx

```diff
@@ -0,0 +1,67 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   TH2XlineReductor.cxx
+/// \author Ivan Ravasenga on the model from Piotr Konopka
+///
+
+#include <TH2.h>
+#include <TMath.h>
+#include "Common/TH2XlineReductor.h"
+
+namespace o2::quality_control_modules::common
+{
+
+void* TH2XlineReductor::getBranchAddress()
+{
+  return &mStats;
+}
+
+/*std::vector<Double_t> TH2XlineReductor::getMean()
+{
+  return mStats.mean;
+}*/
+
+const char* TH2XlineReductor::getBranchLeafList()
+{
+  return " ";
```

Comment:
Sorry, I am not very fluent with TTrees and TBranches. That can work? If you leave a space in the leaf list, it will somehow inspect this structure with vectors? The TTree will correctly store the ever-growing vectors?

Reply:
This was something I forgot to remove. Actually I'm not using getBranchLeafList() for this kind of reductor because of the std::vector (look at lines 55-61 of TrendingTaskITSFhr.cxx). I didn't find any solution to give a leaf list with std::vectors.

Reply:
OK. This is not very clean, but I don't know myself if one can use std::vectors in leaf list. Though, i am pretty sure you can use an array of variable size ( [see "An double array with variable size"](https://root.cern.ch/doc/master/classTTree.html). So perhaps you could use vector.data() and vector.size() there. I guess that would be better than breaking the contract of Reductor.

Reply:
I try to see what I can do. I will be back to you in the next days since I'm also busy with other things.

Reply:
fixed